app/infrastructure/external/email/encryption.py

```
@@ -0,0 +1,57 @@
+    def decrypt(self, encrypted_str: str) -> dict[str, Any]:
+        """Decrypt stored string back to credentials dict.
+
+        Raises:
+            ValueError: If invalid or not valid JSON.
+        """
+        try:
+            decrypted_bytes = self._fernet.decrypt(encrypted_str.encode())
+            json_str = decrypted_bytes.decode()
+            result = json.loads(json_str)
+            if not isinstance(result, dict):
+                raise ValueError("Decrypted credentials must be a dictionary")
```

Ruff (0.14.14)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)